pixeltable/functions/video.py

```diff
@@ -936,6 +939,495 @@ def _create_drawtext_params(
+            frequency data, 4 means only keep 1/4, etc. Larger values make the
+            detector less sensitive to high-frequency details and noise.
+        min_scene_len: Once a cut is detected, this many frames must pass before a new one can be added to the scene
+            list.
+
+    Returns:
+        A list of timestamps (in seconds) where scene cuts were detected, ordered chronologically.
+        If no scene cuts are detected, returns an empty list.
+
+    Examples:
+        Detect scene cuts with default parameters:
+
+        >>> tbl.select(tbl.video.scene_detect_hash()).collect()
+
+        Detect more scenes by lowering the threshold:
+
+        >>> tbl.select(tbl.video.scene_detect_hash(threshold=0.3)).collect()
+
+        Use larger hash size for more precision:
+
+        >>> tbl.select(tbl.video.scene_detect_hash(size=32)).collect()
+
+        Use for fast processing with lower frame rate:
+
+        >>> tbl.select(
+        ...     tbl.video.scene_detect_hash(
+        ...         fps=1.0,
+        ...         threshold=0.4
+        ...     )
+        ... ).collect()
+
+        Add scene cuts as a computed column:
+
+        >>> tbl.add_computed_column(
+        ...     scene_cuts=tbl.video.scene_detect_hash()
+        ... )
+    """
+    from scenedetect.detectors import HashDetector
+
+    try:
+        detector = HashDetector(threshold=threshold, size=size, lowpass=lowpass, min_scene_len=min_scene_len)
+        return _scene_detect(video, fps, detector)
+    except Exception as e:
+        raise pxt.Error(f'scene_detect_hash(): failed to detect scenes: {e}') from e
+
+
+class _SceneDetectFrameInfo(NamedTuple):
+    frame_idx: int
+    frame_pts: int
+    frame_time: float
+
+
+def _scene_detect(video: str, fps: float, detector: 'SceneDetector') -> list[dict[str, int | float]]:
+    from scenedetect import FrameTimecode  # type: ignore[import-untyped]
+
+    with av_utils.VideoFrames(pathlib.Path(video), fps=fps) as frame_iter:
+        video_fps = float(frame_iter.video_framerate)
+
+        scene_cuts: list[dict[str, int | float]] = []
+        frame_idx: int | None = None
+        start_time: float | None = None  # of current scene
+        start_pts: int | None = None  # of current scene
+
+        # in order to determine the cut frame times, we need to record frame times (chronologically) and look them
+        # up by index; trying to derive frame times from frame indices isn't possible due to variable frame rates
+        frame_info: list[_SceneDetectFrameInfo] = []
+
+        def process_cuts(cuts: list[FrameTimecode]) -> None:
+            nonlocal frame_info, start_time, start_pts
+            for cut_timecode in cuts:
+                cut_frame_idx = cut_timecode.get_frames()
+                info_offset = next((i for i, info in enumerate(frame_info) if info.frame_idx == cut_frame_idx), None)
+                assert info_offset is not None
+                info = frame_info[info_offset]
```

Comment:
Add an
```
assert info.frame_time > start_time
```
?

This logic clearly assumes that pyscenedetect presents its scene boundaries in chronological order, and I'd certainly expect that to be the case, but perhaps it's worth adding this assertion just to be safe.

(Or skip over the cut instead of asserting, and log a warning, if we want to be resilient; I'd argue this is better than spitting out a negative duration)